infrastructure/src/main/java/org/corfudb/infrastructure/AbstractServer.java

```diff
@@ -38,14 +38,26 @@ public void sealServerWithEpoch(long epoch) {
 
     public abstract boolean isServerReadyToHandleMsg(CorfuMsg msg);
 
+    /**
+     * A stub that handlers can override to manage their threading, otherwise
+     * the requests will be executed on the IO threads
+     * @param msg
+     * @param ctx
+     * @param r
+     */
+    void processRequest(CorfuMsg msg, ChannelHandlerContext ctx, IServerRouter r) {
```

Comment:
![Codacy](https://app.codacy.com/assets/images/favicon.png) Issue found: [Use explicit scoping instead of the default package private level](https://app.codacy.com/app/annymartinez.m/CorfuDB/pullRequest?prid=4662121)